File: 1146-GreatestCommonDivisorOfStrings/1146-GreatestCommonDivisorOfStrings.py
# Last updated: 4/29/2026, 10:25:45 AM
class Solution:
    def gcdOfStrings(self, str1: str, str2: str) -> str:
        arr = []
        for i in range(min(len(str1), len(str2))):
            if str1[i] == str2[i]:
                arr.append(str1[i])
            else:
                break;
        
        print(arr)
        s = "".join(arr)
        len1 = len(str1)
        len2 = len(str2)
        
        while len(s) >= 1:    
            print(s)
            len_c = len(s)
            p = len1 // len_c
            q = len2 // len_c
            if s * p == str1 and s * q == str2:
                return s
            else:
                s = s[0:len(s)-1]
        return ""

File: 1146-GreatestCommonDivisorOfStrings/1146-GreatestCommonDivisorOfStrings.js
// Last updated: 4/29/2026, 10:25:41 AM
/**
 * @param {string} str1
 * @param {string} str2
 * @return {string}
 */
var gcdOfStrings = function(str1, str2) {
    const n1 = str1.length;
    const n2 = str2.length;
    const half2 = Math.floor(n2 / 2);
    let gcd = '';
    let i = 1;
    for(let i = 1; i <= n2; i++) {
        const candidate = str2.slice(0, i);
        if(n2 % i === 0 && n1 % i === 0) {
            const q1 = n1 / i;
            const q2 = n2 / i;
            if(candidate.repeat(q2) === str2 && candidate.repeat(q1) === str1) {
                gcd = candidate;
            }
        }
    }
    return gcd;
};